src/sync.js

```diff
@@ -635,7 +648,10 @@ SyncApi.prototype._sync = function(syncOptions) {
             self._sync(syncOptions);
         });
         self._currentSyncRequest = null;
-        self._updateSyncState("RECONNECTING");
+        // Transition from RECONNECTING to ERROR after a given number of failed syncs
+        self._updateSyncState(
+            failedSyncCount > FAILED_SYNC_ERROR_THRESHOLD ? "ERROR" : "RECONNECTING",
```

Comment:
Minor, but this probably wants to be >= rather than > so the ERROR state is triggered after FAILED_SYNC_ERROR_THRESHOLD failed syncs which is probably just a bit more intuitive than after n + 1 failed syncs which is what this does.